build/release.mk

```diff
@@ -117,114 +145,186 @@ endif
+			cp tmpprepackaged/$$plugin_package-$$ARCH.tar.gz $(DIST_PATH_LIN_ARM64)/prepackaged_plugins; \
+			cp tmpprepackaged/$$plugin_package-$$ARCH.tar.gz.sig $(DIST_PATH_LIN_ARM64)/prepackaged_plugins; \
+			HAS_ARCH=`tar -tf $(DIST_PATH_LIN_ARM64)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz | grep -oE "dist/plugin-.*"`; \
+			if [ "$$HAS_ARCH" != "dist/plugin-linux-arm64" ]; then \
+				echo "Contains $$HAS_ARCH in $$plugin_package-$$ARCH.tar.gz but needs dist/plugin-linux-arm64"; \
+				exit 1; \
+			fi; \
+			gpg --verify $(DIST_PATH_LIN_ARM64)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz.sig $(DIST_PATH_LIN_ARM64)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz; \
+			if [ $$? -ne 0 ]; then \
+				echo "Failed to verify $$plugin_package-$$ARCH.tar.gz|$$plugin_package-$$ARCH.tar.gz.sig"; \
+				exit 1; \
+			fi; \
+		fi; \
+	done
+	@# Package
+	tar -C $(DIST_PATH_LIN_ARM64)/.. -czf $(DIST_PATH)-$(BUILD_TYPE_NAME)-linux-arm64.tar.gz mattermost ../mattermost
 	@# Preserve native package so dev machines will have an unzipped package available
-	mv $(DIST_PATH_LIN)/bin $(DIST_PATH)/bin
 	@# Cleanup
-	rm -rf $(DIST_PATH_LIN)
+	rm -rf $(DIST_ROOT)/linux_arm64
+
+package-windows: package-prep
+	@# Create needed directories
+	mkdir -p $(DIST_PATH_WIN)/bin
+	mkdir -p $(DIST_PATH_WIN)/logs
+	mkdir -p $(DIST_PATH_WIN)/prepackaged_plugins
+
+	@# Copy binary
+ifeq ($(BUILDER_GOOS_GOARCH),"windows_amd64")
+	cp $(GOBIN)/mattermost.exe $(DIST_PATH_WIN)/bin # from native bin dir, not cross-compiled
+else
+	cp $(GOBIN)/windows_amd64/mattermost.exe $(DIST_PATH_WIN)/bin # from cross-compiled bin dir
+endif
+	#Download MMCTL for Windows
+	scripts/download_mmctl_release.sh "Windows" $(DIST_PATH_WIN)/bin
+	@# Prepackage plugins
+	@for plugin_package in $(PLUGIN_PACKAGES) ; do \
+		ARCH="windows-amd64"; \
+		cp tmpprepackaged/$$plugin_package-$$ARCH.tar.gz $(DIST_PATH_WIN)/prepackaged_plugins; \
+		cp tmpprepackaged/$$plugin_package-$$ARCH.tar.gz.sig $(DIST_PATH_WIN)/prepackaged_plugins; \
+		HAS_ARCH=`tar -tf $(DIST_PATH_WIN)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz | grep -oE "dist/plugin-.*"`; \
+		if [ "$$HAS_ARCH" != "dist/plugin-windows-amd64.exe" ]; then \
+			echo "Contains $$HAS_ARCH in $$plugin_package-$$ARCH.tar.gz but needs dist/plugin-windows-amd64.exe"; \
+			exit 1; \
+		fi; \
+		gpg --verify $(DIST_PATH_WIN)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz.sig $(DIST_PATH_WIN)/prepackaged_plugins/$$plugin_package-$$ARCH.tar.gz; \
+		if [ $$? -ne 0 ]; then \
+			echo "Failed to verify $$plugin_package-$$ARCH.tar.gz|$$plugin_package-$$ARCH.tar.gz.sig"; \
+			exit 1; \
+		fi; \
+	done
+	@# Package
+	cd $(DIST_PATH_WIN)/.. && zip -9 -r -q -l ../mattermost-$(BUILD_TYPE_NAME)-windows-amd64.zip mattermost ../mattermost && cd ../..
+	@# Cleanup
+	rm -rf $(DIST_ROOT)/windows
 
-package: package-osx package-windows package-linux
+package: package-osx-amd64 package-osx-arm64 package-linux-amd64 package-linux-arm64 package-windows
```

Comment:
0/5, but would prefer to keep these targets unchanged, and call these individual targets from under the os level target.